src/v1/item/typers/leaf/validate.ts

```diff
@@ -72,18 +74,16 @@ export const validateLeaf = <L extends Leaf>(
 ): void => {
   validatePropertyState(leafInstance, path)
 
+  const typeValidator = validatorsByLeafType[expectedType]
+
   enumValues?.forEach(enumValue => {
-    if (typeof enumValue !== expectedType) {
+    if (!typeValidator(enumValue)) {
       throw new InvalidEnumValueTypeError({ expectedType, enumValue, path })
     }
   })
 
-  if (
-    defaultValue !== undefined &&
-    defaultValue !== ComputedDefault &&
-    typeof defaultValue !== 'function'
-  ) {
-    if (typeof defaultValue !== expectedType) {
+  if (defaultValue !== undefined && defaultValue !== ComputedDefault && !isFunction(defaultValue)) {
```

Comment:
would extract this to a const to explain the intention of this,
if we ever encounter issues with this piece of code, it would be difficult to understand what's going on here imo

Reply:
Alright 👍 `hasStaticInitialDefaultValue` ?

Initial = is not computed from other inputs - I'm not convinced about this wording. Maybe dependent / independent would be better ?
Static = is not the output of a function

Reply:
Not sure about the wording,
we can just keep it for the last iterations of v1 honestly